qt/scientific_interfaces/Indirect/IndirectDataAnalysisTab.cpp

```diff
@@ -116,19 +150,190 @@ void IndirectDataAnalysisTab::plotCurrentPreview() {
 
     if (inputWs && previewWs->getName() == inputWs->getName()) {
       IndirectTab::plotSpectrum(QString::fromStdString(previewWs->getName()),
-                                m_selectedSpectrum);
+                                m_selectedSpectra);
     } else {
       IndirectTab::plotSpectrum(QString::fromStdString(previewWs->getName()), 0,
                                 2);
     }
   } else if (inputWs &&
              inputWs->getNumberHistograms() <
-                 boost::numeric_cast<size_t>(m_selectedSpectrum)) {
+                 boost::numeric_cast<size_t>(m_selectedSpectra)) {
     IndirectTab::plotSpectrum(QString::fromStdString(inputWs->getName()),
-                              m_selectedSpectrum);
+                              m_selectedSpectra);
+  }
+}
+
+/**
+ * Plots the selected spectrum of the input workspace in this indirect data
+ * analysis tab.
+ */
+void IndirectDataAnalysisTab::plotInput(
+    MantidQt::MantidWidgets::PreviewPlot *previewPlot) {
+  previewPlot->clear();
+  auto inputWS = inputWorkspace();
+
+  if (inputWS)
+    previewPlot->addSpectrum("Sample", inputWorkspace(), selectedSpectra());
+}
+
+/**
+ * Plots the workspace at the specified index in the specified workspace
+ * group. Plots the sample and fit spectrum in the specified top preview
+ * plot. Plots the diff spectra in the specified bottom preview plot.
+ *
+ * @param outputWS          The output workspace group.
+ * @param index             The index of the workspace (in the group)
+ *                          to plot.
+ * @param topPreviewPlot    The top preview plot.
+ * @param bottomPreviewPlot The bottom preview plot.
+ */
+void IndirectDataAnalysisTab::updatePlot(
+    WorkspaceGroup_sptr outputWS, size_t index,
+    MantidQt::MantidWidgets::PreviewPlot *topPreviewPlot,
```

Comment:
Given that the two plots have a specific purpose (the top is for sample and fit spectra, the bottom is for diffraction (?) spectra), could these have more descriptive names?